### PR TITLE
Change types of 'from' and 'to' to 'real' in NumberSelectorDialog and…

### LINKED ISSUE #1442: GUI V2 3.50~15 Lynx Shunt peukert exponent still cannot be changed
see https://victron.discourse.group/t/gui-v2-3-50-15-lynx-shunt-peukert-exponent-still-cannot-be-changed/1251/3

### PR DESCRIPTION
**Pull request by DanielMcInnes**
… ListSpinBox

Fixes #1442

### CODE REVIEW

**Comment by DanielMcInnes** [comment]
The peukert exponent has a 'from' value of 1 and a 'to' value of 1.5. The 'to' and 'from' values were `int`s, and `to: 1.5` got rounded down to 1. If the actual value was 1.2, any change to the value would leave the new value outside of its allowed range, so changes were not enabled.

File: components/listitems/ListSpinBox.qml [hunk 1]
```diff
@@ -13,8 +13,8 @@ ListButton {
 	property alias value: rangeModel.value
 	property string suffix
 	property int decimals
-	property int from: !isNaN(dataItem.min) ? dataItem.min : 0
-	property int to: !isNaN(dataItem.max) ? dataItem.max : Global.int32Max / Math.pow(10, decimals) // qml int is a signed 32 bit value
+	property real from: !isNaN(dataItem.min) ? dataItem.min : 0
+	property real to: !isNaN(dataItem.max) ? dataItem.max : Global.int32Max / Math.pow(10, decimals) // qml int is a signed 32 bit value
```
Comment (chriadam) [inline components/listitems/ListSpinBox.qml:17]: just checking: does the RangeModel below also accept real minimumValue and maximumValue?

Comment (DanielMcInnes) [reply components/listitems/ListSpinBox.qml:17]: yes it does
```
class QRangeModel : public QObject
{
...
	Q_PROPERTY(qreal minimumValue READ minimum WRITE setMinimum NOTIFY minimumChanged FINAL)
	Q_PROPERTY(qreal maximumValue READ maximum WRITE setMaximum NOTIFY maximumChanged FINAL)
```

**Review by blammit — APPROVED** [review]
LGTM, might be worth updating the components demo as well? E.g.

```
diff --git a/pages/settings/debug/PageSettingsDemo.qml b/pages/settings/debug/PageSettingsDemo.qml
index fb8ebb14..830336a6 100644
--- a/pages/settings/debug/PageSettingsDemo.qml
+++ b/pages/settings/debug/PageSettingsDemo.qml
@@ -212,11 +212,11 @@ Page {
 
                        ListSpinBox {
                                text: "Spin box"
-                               value: 5.789
+                               value: 1.2
                                decimals: 2
                                stepSize: Math.pow(10, -decimals)
-                               from: 5
-                               to: 10
+                               from: 1
+                               to: 1.5
                        }
 
                        ListDateSelector {
```

**Comment by DanielMcInnes** [comment]
> LGTM, might be worth updating the components demo as well? E.g.
> 
> ```
> diff --git a/pages/settings/debug/PageSettingsDemo.qml b/pages/settings/debug/PageSettingsDemo.qml
> index fb8ebb14..830336a6 100644
> --- a/pages/settings/debug/PageSettingsDemo.qml
> +++ b/pages/settings/debug/PageSettingsDemo.qml
> @@ -212,11 +212,11 @@ Page {
>  
>                         ListSpinBox {
>                                 text: "Spin box"
> -                               value: 5.789
> +                               value: 1.2
>                                 decimals: 2
>                                 stepSize: Math.pow(10, -decimals)
> -                               from: 5
> -                               to: 10
> +                               from: 1
> +                               to: 1.5
>                         }
>  
>                         ListDateSelector {
> ```

Thanks Bea, updated as suggested

**Pull request merged**